File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/3.4 OrderProcessor Refactor/After/OrderService.java
package com.grokkingsoftwarearchitecture.chapter03.orderprocessor.after;

public class OrderService {
    private final OrderValidator validator;
    private final PaymentService paymentService;
    private final InventoryManager inventoryManager;
    private final NotificationService notificationService;

    public OrderService(OrderValidator validator, PaymentService payment, 
                        InventoryManager inventory, NotificationService notifier) {
        this.validator = validator;
        this.paymentService = payment;
        this.inventoryManager = inventory;
        this.notificationService = notifier;
    }

    public String processOrder(Order order) {
        validator.validate(order);

        if (paymentService.processPayment(order)) {
            inventoryManager.updateInventory(order);
            notificationService.sendConfirmationEmail(order);
            return "Order processed successfully.";
        } else {
            return "Payment failed.";
        }
    }
}

File: Chapter 03/C#/3.4 OrderProcessor Refactor/After/OrderService.cs
namespace Chapter03.OrderProcessorRefactor.After;

public class OrderService
{
    private readonly OrderValidator _validator;
    private readonly PaymentService _paymentService;
    private readonly InventoryManager _inventoryManager;
    private readonly NotificationService _notificationService;

    public OrderService(
       OrderValidator validator, PaymentService payment, 
       InventoryManager inventory, NotificationService notifier)
    {
        _validator = validator;
        _paymentService = payment;
        _inventoryManager = inventory;
        _notificationService = notifier;
    }

    public string ProcessOrder(Order order)
    {
        _validator.Validate(order);

        if (_paymentService.ProcessPayment(order))
        {
            _inventoryManager.UpdateInventory(order);
            _notificationService.SendConfirmationEmail(order);
            return "Order processed successfully.";
        }
        else
        {
            return "Payment failed.";
        }
    }
}